### PR TITLE
[RFR] add all search bar and working well

### PR DESCRIPTION
**Pull request by BaptistePDJ**

### CODE REVIEW

File: src/Controller/AdController.php [hunk 1]
```diff
@@ -25,6 +25,23 @@ public function index(AdRepository $adRepository): Response
...
+        if ("" !== $ids[0]) {
+            foreach ($ids as $id) {
+                $ads[] = $adRepository->findOneBy(['id' => $id]);
+            }
+        }
+
+        return $this->render("ad/search.html.twig", [
+            'ads' => $ads,
+        ]);
+
```
Comment (BaptistePDJ) [inline src/Controller/AdController.php]: BL

File: src/Controller/ClientController.php [hunk 2]
```diff
@@ -15,21 +15,30 @@
...
     #[Route('/', name: 'client_index', methods: ['GET'])]
-    public function index(ClientRepository $clientRepository, EntityManagerInterface $entityManager): Response
+    public function index(Request $request, ClientRepository $clientRepository, EntityManagerInterface $entityManager): Response
     {
         $clients = $clientRepository->findAll();
 
         return $this->render('client/index.html.twig', [
             'clients' => $clients,
             'currentClient' => $clients[0],
+            "error" => $request->get('error') ?? null,
```
Comment (BaptistePDJ) [inline src/Controller/ClientController.php]: still useful ?

File: src/Controller/DomainNameController.php [hunk 3]
```diff
@@ -15,21 +15,30 @@
...
-    public function index(DomainNameRepository $domainNameRepository): Response
+    public function index(Request $request, DomainNameRepository $domainNameRepository): Response
     {
         $domainNames = $domainNameRepository->findAll();
 
         return $this->render('domain_name/index.html.twig', [
             'domain_names' => $domainNames,
-            'currentDomainName' => $domainNames[0] ?? null
+            'currentDomainName' => $domainNames[0] ?? null,
+            "error" => $request->get('error') ?? null,
```
Comment (BaptistePDJ) [inline src/Controller/DomainNameController.php]: same here

File: src/Controller/MailController.php [hunk 4]
```diff
@@ -25,6 +25,23 @@ public function index(MailRepository $mailRepository): Response
...
+        if ("" !== $ids[0]) {
+            foreach ($ids as $id) {
+                $mails[] = $mailRepository->findOneBy(['id' => $id]);
+            }
+        }
+
+        return $this->render("mail/search.html.twig", [
+            'mails' => $mails,
+        ]);
+
```
Comment (BaptistePDJ) [inline src/Controller/MailController.php]: BL

File: templates/ad/index.html.twig [hunk 5]
```diff
@@ -61,7 +98,7 @@
                     </tbody>
                 </table>
             {% else %}
-                <p>Aucune Ad</p>
+                <p>Aucun nom de domaines</p>
```
Comment (BaptistePDJ) [inline templates/ad/index.html.twig]: nom de domaines -> Publicité

File: templates/site/index.html.twig [hunk 6]
```diff
@@ -14,18 +14,45 @@
...
+        function searchSame() {
+            searchItems = [];
+            {% for site in sites %}
+            var siteName = "{{ site.name }}"
+            if (!siteName.includes($('#search').val())) {
+                $('#site-{{ site.id }}').addClass("d-none");
+            } else {
+                $('#site-{{ site.id }}').removeClass("d-none");
+                searchItems.push({{ site.id }});
+            }
```
Comment (BaptistePDJ) [inline templates/site/index.html.twig]: indentation

File: src/Controller/AdController.php [hunk 7]
```diff
@@ -25,6 +25,22 @@ public function index(AdRepository $adRepository): Response
         ]);
     }
 
+    #[Route('/search', name: 'ad_search', methods: ['GET'])]
+    public function search(Request $request, AdRepository $adRepository): Response
+    {
+        $ads= [];
+        $ids = explode(",", $request->get('ids'));
```
Comment (julienmilazzo) [inline src/Controller/AdController.php]:
```
$ids = array_filter(explode(",", $request->get('ids')));
```

File: src/Controller/ClickAndCollectController.php [hunk 8]
```diff
@@ -28,8 +28,16 @@ public function index(ClickAndCollectRepository $clickAndCollectRepository): Res
     #[Route('/search', name: 'click_and_collect_search', methods: ['GET'])]
     public function search(Request $request, ClickAndCollectRepository $clickAndCollectRepository): Response
     {
-        return $this->render('click_and_collect/search.html.twig', [
-            'clickAndCollects' => $clickAndCollectRepository->findBy(['url' => $request->get('searchUrl')]),
+        $clickAndCollects= [];
+        $ids = explode(",", $request->get('ids'));
```
Comment (julienmilazzo) [inline src/Controller/ClickAndCollectController.php]: same here

File: src/Controller/ClientController.php [hunk 9]
```diff
@@ -28,8 +28,16 @@ public function index(ClientRepository $clientRepository, EntityManagerInterface
     #[Route('/search', name: 'client_search', methods: ['GET'])]
     public function search(Request $request, ClientRepository $clientRepository): Response
     {
-        return $this->render('client/search.html.twig', [
-            'clients' => $clientRepository->findBy(['name' => $request->get('searchName')]),
+        $clients = [];
+        $ids = explode(",", $request->get('ids'));
```
Comment (julienmilazzo) [inline src/Controller/ClientController.php]: same here

File: src/Controller/DomainNameController.php [hunk 10]
```diff
@@ -15,21 +15,29 @@
...
+            'currentDomainName' => $domainNames[0] ?? null,
         ]);
     }
 
     #[Route('/search', name: 'domain_name_search', methods: ['GET'])]
     public function search(Request $request, DomainNameRepository $domainNameRepository): Response
     {
+        $domainNames = [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/DomainNameController.php]: same here

File: src/Controller/FrenchEchoppeController.php [hunk 11]
```diff
@@ -28,8 +28,16 @@ public function index(FrenchEchoppeRepository $frenchEchoppeRepository): Respons
     #[Route('/search', name: 'french_echoppe_search', methods: ['GET'])]
     public function search(Request $request, FrenchEchoppeRepository $frenchEchoppeRepository): Response
     {
+        $frenchEchoppes= [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/FrenchEchoppeController.php]: same here

File: src/Controller/MailController.php [hunk 12]
```diff
@@ -25,6 +25,22 @@ public function index(MailRepository $mailRepository): Response
         ]);
     }
 
+    #[Route('/search', name: 'mail_search', methods: ['GET'])]
+    public function search(Request $request, MailRepository $mailRepository): Response
+    {
+        $mails= [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/MailController.php]: same here

File: src/Controller/ServerController.php [hunk 13]
```diff
@@ -27,8 +27,16 @@ public function index(ServerRepository $serverRepository): Response
     #[Route('/search', name: 'server_search', methods: ['GET'])]
     public function search(Request $request, ServerRepository $serverRepository): Response
     {
+        $servers = [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/ServerController.php]: same here

File: src/Controller/SiteController.php [hunk 14]
```diff
@@ -32,8 +32,16 @@ public function index(SiteRepository $siteRepository, EntityManagerInterface $en
     #[Route('/search', name: 'site_search', methods: ['GET'])]
     public function search(Request $request, SiteRepository $siteRepository): Response
     {
+        $sites = [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/SiteController.php]: same here

File: src/Controller/SocialNetworkController.php [hunk 15]
```diff
@@ -25,6 +25,22 @@ public function index(SocialNetworkRepository $socialNetworkRepository): Respons
         ]);
     }
 
+    #[Route('/search', name: 'social_network_search', methods: ['GET'])]
+    public function search(Request $request, SocialNetworkRepository $socialNetworkRepository): Response
+    {
+        $socialNetworks= [];
+        $ids = explode(",", $request->get('ids'));
+        if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/SocialNetworkController.php]: same here

File: templates/mail/search.html.twig [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+                        <td>{{ mail.renewalDate ? mail.renewalDate|date('Y-m-d') : '' }}</td>
+                        <td>{{ mail.commentary }}</td>
+                        <td>{{ mail.enable ? 'Oui' : 'Non' }}</td>
+                        <td>
+                            <a href="{{ path('mail_show', {'id': mail.id}) }}" class="btn btn-secondary">Détails</a>
+                        </td>
+                    </tr>
+                {% else %}
+                    <tr>
+                        <td colspan="8">Aucun Mail
```
Comment (julienmilazzo) [inline templates/mail/search.html.twig]: Where is the end ????? (`</td></tr>`)

File: src/Controller/AdController.php [hunk 17]
```diff
@@ -29,7 +29,7 @@ public function index(AdRepository $adRepository): Response
     public function search(Request $request, AdRepository $adRepository): Response
     {
         $ads= [];
-        $ids = explode(",", $request->get('ids'));
+        $ids = array_filter(explode(",", $request->get('ids')));
         if ("" !== $ids[0]) {
```
Comment (julienmilazzo) [inline src/Controller/AdController.php]: remove the `if`, with array_filter, it's not necessary because of the array_filter

**Pull request merged**